File: Searching/Python/Kth Missing Positive Number in a Sorted Array.py
class Solution:
    def kthMissing(self, arr, k):
        # code here
        lo = 0
        hi = len(arr) - 1
        res = len(arr) + k
        while lo <= hi:
            mid = lo + (hi - lo) // 2
            if arr[mid] > mid + k:
                res = mid + k
                hi = mid - 1
            else :
                lo = mid + 1
        return res

File: Searching/C++/Kth Missing Positive Number in a Sorted Array.cpp
class Solution {
  public:
    int kthMissing(vector<int> &arr, int k) {
        // code here
        int  lo = 0;
        int hi = arr.size() - 1;
        int res = arr.size() + k;
        
        while (lo <= hi) {
            int mid = lo + (hi - lo) / 2;
            if (arr[mid] > mid + k) {
                res = mid + k;
                hi = mid - 1;
            }
            else {
                lo = mid + 1;
            }
        }
        return res;
    }
};
